src/Application/Console/Formatters/CodeClimate.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Application\Console\Formatters;
+
+use Exception;
+use NunoMaduro\PhpInsights\Application\Console\Contracts\Formatter;
+use NunoMaduro\PhpInsights\Domain\Contracts\HasDetails;
+use NunoMaduro\PhpInsights\Domain\Details;
+use NunoMaduro\PhpInsights\Domain\DetailsComparator;
+use NunoMaduro\PhpInsights\Domain\Insights\Insight;
+use NunoMaduro\PhpInsights\Domain\Insights\InsightCollection;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @internal
+ */
+final class CodeClimate implements Formatter
+{
+    private OutputInterface $output;
+
+    public function __construct(OutputInterface $output)
+    {
+        $this->output = $output;
+    }
+
+    /**
+     * Format the result to the desired format.
+     *
+     * @param array<int, string> $metrics
+     *
+     * @throws Exception
+     */
+    public function format(InsightCollection $insightCollection, array $metrics): void
+    {
+        $data = $this->issues($insightCollection, $metrics);
+
+        $this->output->write(json_encode($data));
```

Comment:
may you can add the flag to throw errors to avoid the phpstan error  
```suggestion
        $this->output->write(json_encode($data, JSON_THROW_ON_ERROR));
```